src/common/Autocomplete/DesignationAccountAutocomplete.tsx

```diff
@@ -0,0 +1,66 @@
+import { FocusEventHandler, ReactElement } from 'react';
+import {
+  Autocomplete,
+  AutocompleteProps,
+  CircularProgress,
+  TextField,
+  TextFieldProps,
+} from '@mui/material';
+import { useGetDesignationAccountsQuery } from 'src/components/EditDonationModal/EditDonationModal.generated';
+
+interface DesignationAccountAutocompleteProps
+  extends Partial<AutocompleteProps<string, boolean, boolean, false>> {
+  accountListId: string;
+  onBlur?: FocusEventHandler<HTMLDivElement>;
+  textFieldProps?: Partial<TextFieldProps>;
+}
+
+export const DesignationAccountAutocomplete = ({
+  accountListId,
+  onBlur,
+  textFieldProps,
+  ...props
+}: DesignationAccountAutocompleteProps) => {
+  const { data: designationAccountsData, loading: designationAccountsLoading } =
+    useGetDesignationAccountsQuery({
+      variables: {
+        accountListId,
+      },
+    });
+
+  const designationAccounts =
+    designationAccountsData?.designationAccounts?.flatMap(
+      ({ designationAccounts }) => designationAccounts,
+    );
+
+  return (
+    <Autocomplete
+      autoSelect
+      autoHighlight
+      {...props}
+      options={designationAccounts?.map(({ id }) => id) ?? []}
+      getOptionLabel={(accountId): string => {
+        const account = designationAccounts?.find(({ id }) => id === accountId);
```

Comment:
Great use of `.find` so we don't have to loop through all of the items in the array.